app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php

```diff
@@ -141,6 +143,7 @@ public function __construct(
         ProductLinks $productLinks,
         Js $jsHelper,
         Date $dateFilter,
+        LinkResolver $linkResolver,
         CustomOptionFactory $customOptionFactory = null,
```

Comment:
Please add all new dependencies in a backward-compatible manner - similar to other added in this class:
1) new dependency is last
2) `null` default value
3) initialized in constructor when it was not passed

Reply:
I was sure that it should be done this way but in few of my previous PRs it was mentioned that for 2.4 release proper way is without ObjectManager, and I had to change it for such like here. So which is correct version?

Reply:
@orlangur check https://github.com/magento/magento2/pull/26269 as an example

Reply:
Hi @orlangur, also check this one please https://github.com/magento/magento2/commit/4e6f6ae42cc6109a02d35d0363b1cad0bbac66e1 this was commited by Magento core and this is pretty much the same I did here